app/src/util/tools.ts

```diff
@@ -302,8 +302,14 @@ export const waitForConfirmations = async (
   }
 }
 
-export const formatBigNumber = (value: BigNumber, decimals: number, precision = 2): string => {
-  return Number(formatUnits(value, decimals)).toFixed(precision)
+export function formatBigNumber(value: BigNumber, decimals: number, precision = 2): string {
+  return formatNumber(Number(formatUnits(value, decimals)).toFixed(precision))
+}
+
+export function bigNumberToString(value: BigNumber, decimals: number, precision = 2): string {
+  return Number(formatUnits(value, decimals))
+    .toFixed(precision)
+    .toString()
 }
```

Comment:
@KadenZipfel I researched into this more and the problem with just using BigNumber.toNumber() is that value returned will surpass js biggest safe integer limit and display error like this. It is also outlined in this post
![Screenshot 2021-07-07 at 14 48 43](https://user-images.githubusercontent.com/33226956/124761776-70dfb500-df32-11eb-862f-6011c21c13ea.png)
I was digging around a bit and found that it might be worth considering moving to BigNumber.js dependecy since it solves some of the downfalls of the ethers library and it works..https://mikemcl.github.io/bignumber.js/#toN

Reply:
> @KadenZipfel I researched into this more and the problem with just using BigNumber.toNumber() is that value returned will surpass js biggest safe integer limit and display error like this. It is also outlined in this post on etheres [ethers-io/ethers.js#452](https://github.com/ethers-io/ethers.js/issues/452)
> ![Screenshot 2021-07-07 at 14 48 43](https://user-images.githubusercontent.com/33226956/124761776-70dfb500-df32-11eb-862f-6011c21c13ea.png)

Ahhh good find. It looks like that method will include all the numbers after the decimal, when the intended effect should be reduce the decimals, e.g. `new BigNumber(1)` should become `1` when converted to number, as opposed to 1*10**18. So I guess we'll have to create our function with this effect, probably using `formatUnits`, similar to how Gemino had done in `bigNumberToString`, but simply keeping the value as a precise number instead of converting to string.

Reply:
Yeah i was thinking how to go about it I think we should separate formatting of bigNumber into one used for view which will encompass logic for <0.01 and other function which return number and be used for logic based stuff and I was also thinking making them accept multiple types and handling that logic inside the function so they are more flexible?

Reply:
Interesting idea with accepting multiple types!